src/main/java/com/user_service/services/UserAuthenticationService.java

```diff
@@ -89,10 +97,67 @@ public ConfirmSignUpResponseDto confirmSignUp(ConfirmRegisterDto confirmRegister
             return new ConfirmSignUpResponseDto(null, e.awsErrorDetails().errorMessage());
         }
     }
+
+    public ChangePasswordResponseDto changePassword(ChangePasswordDto changePasswordDto) {
+        try {
+            ChangePasswordRequest changePasswordRequest = ChangePasswordRequest.builder()
+                    .accessToken(changePasswordDto.getAccessToken())
+                    .previousPassword(changePasswordDto.getOldPassword())
+                    .proposedPassword(changePasswordDto.getNewPassword())
+                    .build();
+
+            log.info("User password changed");
+            return new ChangePasswordResponseDto(cognitoIdentityProviderClient.changePassword(changePasswordRequest), null);
+        } catch (CognitoIdentityProviderException e) {
+            log.error("Failed to change user password, error: {}", e.awsErrorDetails().errorMessage());
+            return new ChangePasswordResponseDto(null, e.awsErrorDetails().errorMessage());
+        }
+    }
+
+    public ForgotPasswordResponseDto forgotPassword(ForgotPasswordDto forgotPasswordDto) {
+        try {
+            ForgotPasswordRequest forgotPasswordRequest = ForgotPasswordRequest.builder()
+                    .clientId(CLIENT_ID)
+                    .username(forgotPasswordDto.getEmail())
+                    .build();
+
+            log.info("User {} password restart process started", forgotPasswordDto.getEmail());
+            return new ForgotPasswordResponseDto(cognitoIdentityProviderClient.forgotPassword(forgotPasswordRequest), null);
+        } catch (CognitoIdentityProviderException e) {
+            log.error("Failed to start forgot password process, error: {}", e.awsErrorDetails().errorMessage());
+            return new ForgotPasswordResponseDto(null, e.awsErrorDetails().errorMessage());
+        }
+    }
+
+    public ConfirmForgotPasswordResponseDto confirmForgotPassword(ConfirmForgotPasswordDto confirmForgotPasswordDto) {
+        try {
+            ConfirmForgotPasswordRequest confirmForgotPasswordRequest = ConfirmForgotPasswordRequest.builder()
+                    .clientId(CLIENT_ID)
+                    .username(confirmForgotPasswordDto.getEmail())
+                    .confirmationCode(confirmForgotPasswordDto.getConfirmationCode())
+                    .password(confirmForgotPasswordDto.getPassword())
+                    .build();
+
+            log.info("User {} password reseted successfully", confirmForgotPasswordDto.getEmail());
+            return new ConfirmForgotPasswordResponseDto(cognitoIdentityProviderClient.confirmForgotPassword(confirmForgotPasswordRequest), null);
+        } catch (CognitoIdentityProviderException e) {
+            log.error("Failed to reset user password, error: {}", e.awsErrorDetails().errorMessage());
+            return new ConfirmForgotPasswordResponseDto(null, e.awsErrorDetails().errorMessage());
+        }
+    }
+
+    public void authorize(AuthorizeDto authorizeDto) {
```

Comment:
You should return something from the method. For now it seems that true/false should be enough. We can adjust if our needs change in the future.